### PR TITLE
Feature/analytics data platform

### PR DESCRIPTION
**Pull request by radireddy**
*** Draft review request **

Please review ADP module and APP and provide your comments/suggestions

### CODE REVIEW

File: packages/manager/apps/adp/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@ovh-ux/manager-adp-app",
+  "version": "4.0.0",
```
Comment (antleblanc) [inline packages/manager/apps/adp/package.json]:
```suggestion
  "version": "0.0.0",
```

File: packages/manager/modules/adp/package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "@ovh-ux/manager-adp",
+  "version": "1.0.0",
```
Comment (cbourgois) [inline packages/manager/modules/adp/package.json]:
```suggestion
  "version": "0.0.0",
```

Comment (antleblanc) [inline packages/manager/modules/adp/package.json]:
```suggestion
  "version": "0.0.0",
```

File: packages/manager/modules/adp/package.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/manager-adp' --include-filtered-dependencies -- yarn run dev",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-adp' --include-filtered-dependencies -- yarn run dev:watch",
+    "prepare": "yarn run build"
+  },
+  "dependencies": {
+    "bootstrap4": "twbs/bootstrap#v4.0.0",
+    "lodash": "^4.17.11"
+  },
+  "devDependencies": {
+    "@ovh-ux/component-rollup-config": "5.0.0-beta.5",
```
Comment (antleblanc) [inline packages/manager/modules/adp/package.json]:
```suggestion
    "@ovh-ux/component-rollup-config": "^5.0.0-beta.9",
```

File: packages/manager/modules/adp/src/adp.component.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import deploy from './deploy';
+import ADPService from './adp.service';
+import ADPConstants from './adp.constants';
+import controller from './adp.controller';
+
+const moduleName = 'ovhManagerAdpComponent';
+
+angular
+  .module(moduleName, [deploy])
+  .config(($stateProvider) => {
```
Comment (antleblanc) [inline packages/manager/modules/adp/src/adp.component.js]:
```suggestion
  .config(/* @ngInject */($stateProvider) => {
```

File: packages/manager/apps/adp/package.json (hunk 1 above)
Comment (cbourgois) [inline packages/manager/apps/adp/package.json]:
```suggestion
  "version": "0.0.0",
```

File: packages/manager/apps/adp/package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "build": "webpack",
+    "dev": "webpack-dev-server",
+    "dev:watch": "yarn run dev",
+    "start": "lerna exec --stream --scope='@ovh-ux/manager-adp-app' --include-filtered-dependencies -- yarn run build",
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/manager-adp-app' --include-filtered-dependencies -- yarn run dev",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-adp-app' --include-filtered-dependencies -- yarn run dev:watch"
+  },
+  "dependencies": {
+    "@ovh-ux/manager-core": "^4.0.0",
+    "@ovh-ux/manager-adp": "file:../../modules/adp",
```
Comment (cbourgois) [inline packages/manager/apps/adp/package.json]:
```suggestion
    "@ovh-ux/manager-adp": "^0.0.0",
```

File: packages/manager/modules/adp/src/adp.component.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      component: 'ovhManagerAdpComponent',
+      translations: {
+        value: [
+          '.',
+        ],
+        format: 'json',
+      },
+    });
+  })
+  .component('ovhManagerAdpComponent', {
```
Comment (antleblanc) [inline packages/manager/modules/adp/src/adp.component.js]: You can create a dedicated file that contain your component and export it.

```js
import component from './component';
```

File: packages/manager/modules/adp/src/adp.component.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    template,
+    controller,
+    controllerAs: '$ctrl',
+  })
+  .service('AdpService', ADPService)
```
Comment (antleblanc) [inline packages/manager/modules/adp/src/adp.component.js]: Prefer using camelCase for services:

```suggestion
  .service('adpService', ADPService)
```

File: packages/manager/modules/adp/src/adp.constants.js [hunk 8]
```diff
@@ -0,0 +1,251 @@
+export default {
```
Comment (cbourgois) [inline packages/manager/modules/adp/src/adp.constants.js]: Should export each `const` and export an object containing const as `default`.

```js
export const PUBLIC_CLOUD_STATUS = ...
export const ADP_CAPABILITIES = ...

export default {
  PUBLIC_CLOUD_STATUS,
  ADP_CAPABILITIES,
}
```

File: packages/manager/modules/adp/src/deploy/deploy.component.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import template from './deploy.html';
+import controller from './deploy.controller';
+
+const moduleName = 'ovhManagerAdpDeployComponent';
+
+angular
+  .module(moduleName, [
+    'ui.router',
+  ])
+  .config(($stateProvider) => {
```
Comment (antleblanc) [inline packages/manager/modules/adp/src/deploy/deploy.component.js]:
```suggestion
  .config(/* @ngInject */($stateProvider) => {
```

File: packages/manager/modules/adp/src/adp.constants.js [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+    }, {
+      name: 'Apache Zeppelin',
+      version: '0.7.3',
+    }, {
+      name: 'Apache ZooKeeper',
+      version: '3.4.6',
+    }],
+    available_region: ['GRA5', 'SBG5'],
+    bastion_node: {
+      instance_type: ['b2-7'],
```
Comment (cbourgois) [inline packages/manager/modules/adp/src/adp.constants.js]: `s/insatnce/instance`

File: packages/manager/modules/adp/src/adp.service.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
+import filter from 'lodash/filter';
+import isEqual from 'lodash/isEqual';
+import map from 'lodash/map';
+
+export default class ADPService {
+  /* @ngInject */
+
```
Comment (cbourgois) [inline packages/manager/modules/adp/src/adp.service.js]: extra empty line

File: packages/manager/modules/adp/src/deploy/index.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+import nodesTemplate from './nodes.html';
+
+const moduleName = 'ovhManagerAdpDeploy';
+
+angular
+  .module(moduleName, [
+    'ui.router',
+    'oc.lazyLoad',
+  ])
+  .config(($stateProvider) => {
```
Comment (antleblanc) [inline packages/manager/modules/adp/src/deploy/index.js]:
```suggestion
  .config(/* @ngInject */($stateProvider) => {
```

File: packages/manager/modules/adp/src/deploy/index.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    $stateProvider.state('adp.deploy.**', {
```
Comment (cbourgois) [inline packages/manager/modules/adp/src/deploy/index.js]: 👍

File: packages/manager/modules/adp/src/deploy/index.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+      url: '/deploy',
+      lazyLoad: ($transition$) => {
+        const $ocLazyLoad = $transition$.injector().get('$ocLazyLoad');
+        return import('./deploy.component')
+          .then(mod => $ocLazyLoad.inject(mod.default || mod));
+      },
+    });
+  }).run(/* @ngInject */($templateCache) => {
+    // import templates required by ng-include
+    $templateCache.put('adp/deploy/nodes.html', nodesTemplate);
```
Comment (cbourgois) [inline packages/manager/modules/adp/src/deploy/index.js]: 👍

File: packages/manager/modules/adp/src/translations/Messages_en_GB.json [hunk 15]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (cbourgois) [inline packages/manager/modules/adp/src/translations/Messages_en_GB.json]: You shouldn't commit non `fr_FR` translations files

**Comment by cbourgois** [comment]
Great start ✨

File: package.json [hunk 16]
```diff
@@ -61,5 +61,8 @@
   "engines": {
     "node": ">=8.11",
     "yarn": ">=1.12.3"
+  },
+  "dependencies": {
+    "ovh-ui-angular": "^2.25.1"
```
Comment (marie-j) [inline package.json]: this should probably not be in the root `package.json`

File: packages/manager/apps/adp/index.html [hunk 17]
```diff
@@ -0,0 +1,9 @@
+<!doctype html>
+<html>
+    <head>
+        <title>ADP App</title>
+    </head>
+    <body data-ng-app="adpApp">
```
Comment (marie-j) [inline packages/manager/apps/adp/index.html]:
```suggestion
    <body data-ng-app="adpApp" data-ng-strict-di>
```

File: packages/manager/modules/adp/package.json [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/manager-adp' --include-filtered-dependencies -- yarn run dev",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-adp' --include-filtered-dependencies -- yarn run dev:watch",
+    "prepare": "yarn run build"
+  },
+  "dependencies": {
+    "bootstrap4": "twbs/bootstrap#v4.0.0",
+    "lodash": "^4.17.11"
+  },
+  "devDependencies": {
+    "@ovh-ux/component-rollup-config": "5.0.0-beta.9",
```
Comment (marie-j) [inline packages/manager/modules/adp/package.json]:
```suggestion
    "@ovh-ux/component-rollup-config": "^5.0.0-beta.9",
```

File: packages/manager/modules/adp/src/adp.html [hunk 19]
```diff
@@ -0,0 +1,5 @@
+<section class="adp">
+    <div data-ui-view></div>
+</section>
+
+
```
Comment (marie-j) [inline packages/manager/modules/adp/src/adp.html]: Extra empty line

File: packages/manager/modules/adp/src/deploy/deploy.component.js [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+          './..',
+        ],
+        format: 'json',
+      },
+    });
+  })
+  .component('ovhManagerAdpDeployComponent', {
+    template,
+    controller,
+    controllerAs: '$ctrl',
```
Comment (marie-j) [inline packages/manager/modules/adp/src/deploy/deploy.component.js]: `$ctrl` is the default `controller` name for a `component` :) 
```suggestion
```

File: packages/manager/modules/adp/src/deploy/deploy.controller.js [hunk 21]
```diff
@@ -0,0 +1,300 @@
...
+    this.regionsByContinent = this.cucControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.regionsByDatacenter
+        .promise
+        .then(regions => groupBy(regions, 'continent')),
+    });
+    return this.regionsByContinent.load();
+  }
+
+  submitRegionAndDatacenter() {
+    console.log('region', this.selectedRegion);
```
Comment (marie-j) [inline packages/manager/modules/adp/src/deploy/deploy.controller.js]: You should probably remove those `console.log` once the PR will be considered as ready :)

File: packages/manager/modules/adp/src/deploy/deploy.html [hunk 22]
```diff
@@ -0,0 +1,56 @@
+<section>
+    <!-- page header -->
+    <oui-page-header
+        data-heading="Deploy an Analytics Data Platform"
+        data-description="OVH Analytics Data Platform (ADP) contains all the tools to ingest, clean, process and visualize vast amounts data.
+                We deploy a preconfigured and fully secured Apache Hadoop cluster, on top of OVH Public Cloud compute and storage resources.
+                Then, simply upload your data or add data collector for real-time flows, and follow our documentation to make us of ADP benefits.">
+    </oui-page-header>
+    <!-- page content -->
+    <div class="mx-auto" style="max-width: 1200px;">
```
Comment (marie-j) [inline packages/manager/modules/adp/src/deploy/deploy.html]: Could you use Bootstrap utilities instead ? https://getbootstrap.com/docs/4.1/utilities/sizing/

File: packages/manager/modules/adp/src/deploy/regions.html [hunk 23]
```diff
@@ -0,0 +1,45 @@
+<oui-tabs class="mb-xs-2 mb-4">
+    <!-- all regions -->
+    <oui-tabs-item data-heading="{{ 'adp_deploy_all_regions' | translate }}">
+        <div class="d-flex d-sm-flex mx-3" style="flex-wrap: wrap;">
```
Comment (marie-j) [inline packages/manager/modules/adp/src/deploy/regions.html]: https://getbootstrap.com/docs/4.1/utilities/flex/
```suggestion
        <div class="d-flex d-sm-flex flex-wrap mx-3">
```

**Pull request closed without merging**